test/lightning/adaptor_registry_test.exs

```diff
@@ -38,19 +33,26 @@ defmodule Lightning.AdaptorRegistryTest do
     end
 
     test "retrieves a list of adaptors when caching is disabled" do
-      # :hackney.request(request.method, request.url, request.headers, request.body, hn_options)
-      expect(:hackney, :request, fn
-        :get,
-        "https://registry.npmjs.org/-/user/openfn/package",
-        [],
-        "",
-        [recv_timeout: 15_000, pool: :default] ->
-          {:ok, 200, "headers", :client}
-      end)
-
-      expect(:hackney, :body, fn :client, _timeout ->
-        {:ok, File.read!("test/fixtures/openfn-packages-npm.json")}
-      end)
+      default_npm_response =
+        File.read!("test/fixtures/language-common-npm.json") |> Jason.decode!()
+
+      expect_tesla_call(
```

Comment:
Niice!! Never new this existed 🤩

Reply:
Ha! Yeah it's pretty useful, it's a tiny wrapper around Mox.

I'm not in completely in love with it - it is a bit awkward when you may have multiple requests that _could_ be out of order. 

But I found an undocumented value for the `:returns` atom, it can be a function - which allows you to case statement over the different responses you may want depending on the request: https://github.com/elixir-tesla/tesla/blob/v1.13.2/lib/tesla/test.ex#L228